MekHQ/src/mekhq/gui/dialog/GMToolsDialog.java

```diff
@@ -42,887 +39,1256 @@
+            if (doesPersonPrimarilyDriveUnitType(UnitType.determineUnitTypeCode(getComboUnitType().getItemAt(i)))) {
+                getComboUnitType().setSelectedIndex(i);
                 break;
             }
         }
 
         if (!StringUtil.isNullOrEmpty(getPerson().getBloodname())) {
-            currentBloodname.setText(getPerson().getBloodname());
+            getLblCurrentBloodname().setText(getPerson().getBloodname());
         }
 
-        int year = gui.getCampaign().getGameYear();
-        for (int i = ERAS.length - 1; i >= 0; i--) {
-            if (ERAS[i] <= year) {
-                bloodnameEraPicker.setSelectedIndex(i);
-                year = ERAS[i];
+        int year = getGUI().getCampaign().getGameYear();
+        for (int i = BLOODNAME_ERAS.length - 1; i >= 0; i--) {
+            if (BLOODNAME_ERAS[i] <= year) {
+                getComboBloodnameEra().setSelectedIndex(i);
+                year = BLOODNAME_ERAS[i];
                 break;
             }
         }
 
-        originClanPicker.setSelectedItem((gui.getCampaign().getFaction().isClan()
-                ? gui.getCampaign().getFaction() : getPerson().getOriginFaction()).getFullName(year));
-
-        phenotypePicker.setSelectedItem(getPerson().getPhenotype());
-    }
-
-    /**
-     * Finds the initial faction, which is either the person's faction (if
-     * the dialog was launched for a specific person) or the campaign's
-     * faction.
-     */
-    private int findInitialSelectedFaction(List<FactionChoice> factionChoices) {
-        final Faction faction = (getPerson() != null)
-                ? getPerson().getOriginFaction() : getGUI().getCampaign().getFaction();
-        if (faction == null) {
-            return 0;
+        final Clan clan = Clan.getClan((getGUI().getCampaign().getFaction().isClan()
+                ? getGUI().getCampaign().getFaction() : getPerson().getOriginFaction()).getShortName());
+        if (clan != null) {
+            getComboOriginClan().setSelectedItem(new ClanDisplay(clan, getGUI().getCampaign().getLocalDate()));
         }
 
-        int index = 0;
-        for (FactionChoice factionChoice : factionChoices) {
-            if (factionChoice.getFaction().equals(faction)) {
-                return index;
-            }
-
-            index++;
-        }
-
-        return 0;
+        getComboPhenotype().setSelectedItem(getPerson().getPhenotype());
     }
 
     /**
      * Determine if a person's primary role supports operating a
      * given unit type.
      */
-    private boolean doesPersonPrimarilyDriveUnitType(int unitType) {
-        PersonnelRole primaryRole = getPerson().getPrimaryRole();
+    private boolean doesPersonPrimarilyDriveUnitType(final int unitType) {
         switch (unitType) {
             case UnitType.AERO:
-                return primaryRole.isAerospacePilot();
+                return getPerson().getPrimaryRole().isAerospacePilot();
```

Comment:
Not sure I get why we got rid of line 744 and added more code across the rest of the switch.

Reply:
Personal preference, if the line isn't too long then this allows me to tell the whole story in a single line.

Reply:
Does the compiler know that?

Reply:
There shouldn't be any difference in compilation. Actually, I believe this would execute extremely slightly faster because it doesn't need to create a temporary reference that then immediately needs to be garbage collected.

Reply:
In this case, that's reasonable, since it's a switch and not a loop, so only one of these would get executed anyway.